doc/source.rst

```diff
@@ -181,12 +181,16 @@ File patterns
 File path patterns are used for include and omit, and for combining path
 remapping.  They follow common shell syntax:
 
-- ``*`` matches any number of file name characters, not including the directory
-  separator.
-
 - ``?`` matches a single file name character.
 
-- ``**`` matches any number of nested directory names, including none.
+- ``*`` as part of a path component matches any number of file name characters,
+  not including the directory separator: one path component.
+
+- ``*`` as the first or last path component matches anything including
+  directory separators: any number of path components, including none, like
```

Comment:
So this kinda implies that `"*thing/stuff/"` would match `"some/xthing/stuff/"` because it's unclear what a “path component” refers to.
I think, this is because of the ambiguity — the match expression with a wildcard looks like `"*thing/stuff/"` and so I read this as `"*thing"` is the first path component, and it matches things like `[a-zA-Z0-9/]` basically.
So while you must've been thinking of the path of a discovered file during the filtering process, the reader is likely to think of what's in front of them — a file path pattern.

Maybe, we could rephrase this by specifically pointing out the “if a path pattern starts with `*/`, this is equivalent to `**/`”? WDYT?

Reply:
Yeah, I had a few tries at how to explain this, and didn't love any of them.  I think your idea is a good one.